core/controllers/admin.py

```diff
@@ -225,9 +241,9 @@ def _reload_exploration(self, exploration_id):
             logging.info(
                 '[ADMIN] %s reloaded exploration %s' %
                 (self.user_id, exploration_id))
-            exp_services.load_demo(unicode(exploration_id))
+            exp_services.load_demo(str(exploration_id))
```

Comment:
Why str here and builtins.str above?

In general prefer builtins.XXX -- it's clearer.

Also, related to this -- are you maintaining a doc for "instructions for fully migrating to python 3"? Please link to that doc in this PR, since it would be good to see action items like "remove `builtins.` prefix" recorded somewhere stable. This is something the reviewers need to check for (i.e. understanding the entire migration process to python 3, even the part that might need to happen after GSoC).

Reply:
We need to use str here since load_demo finally calls `get_exploration_by_id` and that calls the datastore function `get_by_id` and all the functions in the data store level need to have python 2's str as parameters. Re the doc, I will be sharing it ASAP.

Reply:
OK, maybe I don't understand what's going on. What's the difference between str and builtins.str?

Also, could you implement this as python_utils.to_str() and python_utils.to_unicode() (or python_utils.to_bytes() and python_utils.to_unicode_str(), say)?

Reply:
> Also, could you implement this as python_utils.to_str() and python_utils.to_unicode() (or python_utils.to_bytes() and python_utils.to_unicode_string(), say)?

done
> OK, maybe I don't understand what's going on. What's the difference between str and builtins.str?

builtins.str is of the type `future.types.newstr.newstr` in Python 2 while its of type `str` in Python 3. That's why builtins.str does not work in the datastore level as it provides compatibility issues

Reply:
Re the doc, I think I can make it better once I start with the string migrations -- that way I will know of more changes that needs to be done. Does that sound OK?